### PR TITLE
Fix/12300 made slugify/urlify to strip unrequired characters

### LINKED ISSUE #12300: JS slugify (with unicode enabled) leaves invalid characters in the slug
### Issue Summary

The JS `slugify` function (used when cleaning a manually-entered slug) fails to strip out certain spacer/combining characters that are disallowed by Django slug validation.

### Steps to Reproduce

1. On a site with `WAGTAIL_ALLOW_UNICODE_SLUGS = True`, create a new page
2. Paste the text "উইকিপিডিয়ায় স্বাগতম!" into the slug field
3. Defocus the field; it will be rewritten to "উইকিপিডিয়ায়-স্বাগতম"
4. Attempt to save the page; this returns the error "Enter a valid “slug” consisting of Unicode letters, numbers, underscores, or hyphens."

The expected behaviour is that the browser-side slug cleanup step should result in a slug that is accepted by the Django validation. However, the output contains the character [U+09BF](https://www.fileformat.info/info/unicode/char/09bf/index.htm) which is disallowed, being a spacing combining character.

Note that the bug only arises for slugs entered manually into the slug field; pasting "উইকিপিডিয়ায় স্বাগতম!" into the _title_ field results in the slug "উইকপডযয-সবগতম" which is valid. This is because the title-to-slug conversion uses a more complex function which also performs transliteration when unicode slugs are disabled (which wouldn't be appropriate for manually-entered slugs); however, the simplified function used for cleaning manually-entered slugs is _overly_ simple:

https://github.com/wagtail/wagtail/blob/84b3bf70349171e44f283085f1d89ccd9deed40e/client/src/utils/slugify.ts#L10-L13

since it's only stripping out a designated short list of punctuation characters, where it would be better to strip out everything that is _not_ classed as a letter/number or explicitly allowed punctuation (just hyphen and underscore I think). Would be useful to compare this against the behaviour of the more advanced title-to-slug conversion, and Django's `django.utils.text.slugify`.

-   I have confirmed that this issue can be reproduced as described on a fresh Wagtail project: yes

### Technical details

-   Python version: 3.12.4.
-   Django version: 5.1.1
-   Wagtail version: 6.3a0
-   Browser version: Chrome 128

### Working on this

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

### PR DESCRIPTION
**Pull request by Atif2077**
Added a new function in wagtail/client/utils/slugify.ts which would only pass the values which are allowed by the django's slug validator.
Also added a test case which was given in the issue.
fixes #12300

### CODE REVIEW

File: wagtail/migrations/0001_initial.py [hunk 1]
```diff
@@ -1,291 +0,0 @@
-from django.conf import settings
```
Comment (zerolab) [inline wagtail/migrations/0001_initial.py:1]: please reinstante this file.

Comment (Atif2077) [reply wagtail/migrations/0001_initial.py:1]: Is there anything else... which I can do from my end?

**Comment by gasman** [comment]
Hi @Atif2077 - as @zerolab said, you've accidentally deleted the file `wagtail/migrations/0001_initial.py` in your first commit. You need to add this back (or start again with a new branch based on `main`).

Also, I think your new code can be simplified a bit. `sanitizeSlug` doesn't really need to be a function of its own - it's just one more step in the `slugify` process, and someone reading the code will have an easier time understanding it if they can read all the steps relating to the `allowUnicode` case from top to bottom. The old line ```.replace(/[&/\\#,+()$~%.'":`@^!*?<>{}]/g, '')``` is also not needed now, as your new code is performing the same job of removing punctuation (and more).

I'd recommend turning the `allowUnicode ? (...) : (...)` expression into an `if` statement
```
if (allowUnicode) {
   // ...
} else {
   // ...
}
```
so that your new code can easily go into the first of those blocks, without having to squeeze everything into a single expression.

Thanks again for the PR (and especially thankyou for including a test case - lots of people miss that step out :-) )!

**Comment by Atif2077** [comment]
Thank you so much @gasman and @zerolab for your responses, I will be making the changes in the code to improve it and would also add that line in migrations.

**Comment by Atif2077** [comment]
Hey @gasman and @zerolab I have made the necessary changes which were mentioned above, where can I improve more with this code.

File: wagtail/migrations/0001_initial.py (hunk 1 above)
Comment (zerolab) [reply wagtail/migrations/0001_initial.py]: Can you run `git checkout main -- wagtail/migrations/0001_initial.py` from the command line in the repository folder so the file doesn't appear in the list of changed files at all

Comment (Atif2077) [reply wagtail/migrations/0001_initial.py]: Nothing is happening, I don't know why I tried changing the .py file and again taking it original but still the change is there.
can you briefly explain the steps.
Also, I am editing these on my fix branch.

**Comment by gasman** [comment]
Thanks @Atif2077! Now merged in 96f8324458d389d79435bc018eb98b1af2d93d03.

I've credited you in the release notes as Atif2077 - let me know if you'd like this name changed to something else.

**Comment by Atif2077** [comment]
### Thank you @gasman, would love if you can change my name to Atif Khan and It would be very helpful if you can tell me some technologies which I can learn to contribute better. 

### Currently I know,
- Django with Channels and flask
- A little of other things like JavaScript, Docker, SQL
- I know languages like C, C++, Python

**Pull request closed without merging**